d4s2_api_v2/api.py

```diff
@@ -265,6 +266,19 @@ def _give_agent_permission(s3_delivery, user):
         except S3Exception as ex:
             raise WrappedS3Exception(ex)
 
+    @staticmethod
+    def _record_object_manifest(s3_delivery, user):
+        """
+        Update delivery recording the object manifest based on data in s3
+        :param s3_delivery: S3Delivery: details about what we will deliver
+        :param user: user who's credentials to use to talk to s3
+        """
+        bucket = s3_delivery.bucket
+        s3_bucket_util = S3BucketUtil(bucket.endpoint, user)
+        objects_manifest = s3_bucket_util.get_objects_manifest(bucket_name=bucket.name)
+        s3_delivery.object_manifest = json.dumps(objects_manifest)
```

Comment:
I think the responsibility of serializing this dictionary to/from JSON should be handled by the model. Another reason to use JSONField 😄